File: app/src/main/java/zmuzik/slidingpuzzle2/mainscreen/SavedPicturesGridView.java
package zmuzik.slidingpuzzle2.mainscreen;

import android.content.Context;

import java.util.List;

/**
 * Created by Zbynek Muzik on 2017-03-28.
 */

public class SavedPicturesGridView extends BasePicturesGridView {

    public SavedPicturesGridView(Context context) {
        super(context);
    }

    @Override
    public void requestUpdate() {
        mPresenter.requestUpdateSavedPictures();
    }

    @Override
    public void update(List<String> pictures) {
        mAdapter = new PicturesGridAdapter(getContext(), pictures, getColumnsNumber());
        mRecyclerView.setAdapter(mAdapter);
    }
}

File: app/src/main/java/zmuzik/slidingpuzzle2/mainscreen/SavedPicturesGridView.kt
package zmuzik.slidingpuzzle2.mainscreen

import android.content.Context

/**
 * Created by Zbynek Muzik on 2017-03-28.
 */

class SavedPicturesGridView(context: Context) : BasePicturesGridView(context) {

    override fun requestUpdate() {
        presenter.requestUpdateSavedPictures()
    }
}
